src/db/collections/milestone.rs

```diff
@@ -269,8 +269,35 @@ impl MongoDb {
             )
             .await?
             .try_next()
-            .await?
-            .map(|d| d.milestone_index))
+            .await
+    }
+
+    /// Find the latest inserted milestone in the range the provided milestone index is included.
+    pub async fn get_latest_milestone_in_range(
+        &self,
+        index: MilestoneIndex,
+    ) -> Result<Option<MilestoneIndexTimestamp>, Error> {
+        let mut stream = self
+            .0
+            .collection::<MilestoneIndexTimestamp>(MilestoneDocument::COLLECTION)
+            .find(
+                doc! { "milestone_index": { "$gte": index } },
+                FindOptions::builder().sort(doc! {"milestone_index": 1}).build(),
+            )
```

Comment:
I'm not sure if this is the right collection to get the information from. I think it would be better to have a `ledger_index` field in the `NodeStatus` collection, which we update separately. This way we could also achieve _O(1)_ lookups (instead of _O(n)_ for the `$max`-based approach).

When we start using `ReadUnspentOutputs` we can't really add this information to the `MilestoneDocument` anyway.